src/sections/wrapper.tsx

```diff
@@ -0,0 +1,282 @@
+import { useEffect, useState } from 'react';
+import { useBalance, useContractWrite, usePrepareContractWrite, useAccount } from 'wagmi';
+import { parseUnits } from 'viem';
+import wrapperAbi from '../abi/wrapperAbi.json';
+
+import styled from '@emotion/styled';
+import {
+  IconButton,
+  Paper,
+  TextField,
+  InputAdornment,
+  Button as MuiButton
+} from '@mui/material'
+import Button from '../future-hopr-lib-components/Button';
+import Section from '../future-hopr-lib-components/Section';
+
+import SwapVertIcon from '@mui/icons-material/SwapVert';
+import LaunchIcon from '@mui/icons-material/Launch';
+
+const StyledPaper = styled(Paper)`
+  padding: 2rem;
+  text-align: center;
+`;
+
+const WrapperContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 1rem;
+  position: relative;
+
+  & .swap-button {
+    align-self: center;
+    text-transform: uppercase;
+  }
+`;
+
+const StyledTextField = styled(TextField)`
+  /* Chrome, Safari, Edge, Opera */
+  input::-webkit-outer-spin-button,
+  input::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  /* Firefox */
+  input[type='number'] {
+    -moz-appearance: textfield;
+  }
+`;
+
+const StyledIconButton = styled(IconButton)`
+  align-self: center;
+  position: absolute;
+  background: linear-gradient(rgba(0, 0, 178, 1), rgba(0, 0, 80, 1));
+  top: 48px;
+  z-index: 2;
+
+  & svg {
+    color: #fff;
+  }
+
+  &:disabled {
+    background: #e0e0e0;
+    & svg {
+      color: #a6a6a6;
+    }
+  }
+`;
+
+const MaxButton = styled(MuiButton)`
+  background-color: #ffffa0;
+  border-radius: 2px;
+  border: none;
+  color: #444;
+  font-weight: 600;
+  padding: 0.2rem 1rem;
+
+  &:disabled {
+    background-color: #e0e0e0;
+    color: #a6a6a6;
+  }
+`;
+
+const GnosisLink = styled.a`
+  display: inline-flex;
+  gap: 2px;
+  text-decoration: underline;
+
+  & svg {
+    align-self: flex-end;
+    height: 16px;
+    width: 16px;
+  }
+`;
+
+type NumberLiteral = `${number}`;
+
+type TransactionLinkProps = {
+  isSuccess: boolean;
+  hash: `0x${string}` | undefined;
+};
+
+function TransactionLink({
+  isSuccess,
+  hash,
+}: TransactionLinkProps) {
+  if (!isSuccess) return null;
+
+  return (
+    <span>
+      Check your{' '}
+      <GnosisLink href={`https://gnosisscan.io/tx/${hash}`}>
+        transaction
+        <LaunchIcon />
+      </GnosisLink>
+    </span>
+  );
+}
+
+function WrapperPage() {
+  const [xhoprValue, set_xhoprValue] = useState('');
+  const [wxhoprValue, set_wxhoprValue] = useState('');
+  const [swapDirection, set_swapDirection] = useState<'xHOPR_to_wxHOPR' | 'wxHOPR_to_xHOPR'>('xHOPR_to_wxHOPR');
+  const { address } = useAccount();
+
+  // Fetch balance data
+  const { data: xHOPR_balance } = useBalance({
+    address,
+    token: '0xD057604A14982FE8D88c5fC25Aac3267eA142a08',
```

Comment:
const value for repeated sc addresses

Reply:
Added consts for smart contract addresses